### PR TITLE
Bootstrap collapsible elements do not degrade gracefully

### PR DESCRIPTION
**Pull request by tqureshi-uog**
For #732 - Adding the noscript fix so bootstrap elements with collapse class expand by default if javascript disabled

### CODE REVIEW

**Comment by nevantan** [comment]
This fix works when minimal markup is used:
```html
<button data-toggle="collapse" data-target="#demo">Collapsible</button>
<div id="demo" class="collapse">
    Some content that is hidden when collapsed.
</div>
```

However, Grad Studies (https://www.uoguelph.ca/graduatestudies/future/applying-guelph) specifically seems to have one collapsible area that uses `height:0px;` which causes an overlap:
overlap
Setting `height:auto;` or just removing that inline CSS rule should fix it:
fixed

Additionally, the elements seem to show `aria-expanded="false"` by default. Should these also be changed to reflect their actual state when relying on the fallback?

**Comment by ljairath** [comment]
Please notify the clients once this issue is resolved. These are the two clients that I know of requesting for the feature. Add client to the following list if this feature requested. 
 
Steve Forbes: FP #292387 Re: Drupal question Ticket closed.
Renee Shea: FP #290521 Re: question abour restructuring our HR Contacts Page

**Comment by brharp** [comment]
Can we use a selector on <html class="js"> rather than the <noscript> tag?

**Comment by tqureshi-uog** [comment]
@brharp How would the selector work, exactly? I see the "js" class is applied to the html element when javascript is enabled. Can this address the "aria-expanded" issue Nevan pointed out above?

**Comment by brharp** [comment]
Instead of the noscript tag, we could use the following CSS selector:
```
html[class^=js] .collapse {
    display: block;
}
```

**Comment by nevantan** [comment]
Added a script to correct the aria-expanded attribute when JavaScript is enabled. This will require some CSS solution for showing elements with `.collapse`.

Clients will also need to be notified to update their markup to set `aria-expanded="true"` as they will likely have `aria-expanded="false"` already. Also, clients that would like a section to be expanded by default on page load (with JavaScript enabled) should add the `in` class to the element with `.collapse` as recommended by Bootstrap (see: http://getbootstrap.com/javascript/#collapse note 'Make expand/collapse controls accessible')

**Comment by nevantan** [comment]
**A note on expected behavior:**
When JavaScript is _unavailable_, collapsible sections will _always_ be expanded with `aria-expanded="true"`.

When JavaScript is _available_, collapsible sections will be closed by default and can be opened on click. Adding `.in` to a collapsible section (ie. `class="collapse in"`) will show the section on page load. The `aria-expanded` will correct itself once on page load and remain in sync as sections are toggled.

File: profiles/ug/themes/ug/ug_theme/js/ug_scripts.js [hunk 1]
```diff
@@ -1,53 +1,12 @@
...
-		  }, 10);
+/*-------------------------------------
+Bootstrap collapsible aria-expanded fix
+------------------------------------**/
+jQuery(function($) {
+	$(document).ready(function(e) {
+		$(".collapse").each(function(i, elem) {
+			var isExpanded = $(elem).hasClass('in');
+			$(elem).attr("aria-expanded", isExpanded);
+			$('[data-target="#' + $(elem).attr("id") + '"]').attr("aria-expanded", isExpanded);
```
Comment (brharp) [inline profiles/ug/themes/ug/ug_theme/js/ug_scripts.js]: I don't think we need aria-expanded on the controlling element

Comment (brharp) [inline profiles/ug/themes/ug/ug_theme/js/ug_scripts.js]: Would it be possible to add aria-controls="id" to the button with data-target="id" or the a with href="#id"? According to the bootstrap docs, data-target is only used with buttons, so I think we would have to account for the case where an a[href] is the controlling element.

Comment (nevantan) [inline profiles/ug/themes/ug/ug_theme/js/ug_scripts.js]: According to the Bootstrap 4 docs, data-target is required for both elements (https://v4-alpha.getbootstrap.com/components/collapse/). We are using an older version or Bootstrap, but it doesn't hurt to have the `data-target` attribute on a link element and will make selectors easier to write.

**Comment by nevantan** [comment]
The JavaScript now selected all `.collapse` elements, loops through them, and selects their associated controlling element using it's `data-target` attribute. With both of these elements selected, they can be manipulated the way we need.

In order to make the markup as simple as possible, any attributes that aren't required will be set automatically by JavaScript. At it's most basic, a collapsible setup looks like this:
```html
<!-- Controlling element to toggle collapsible content -->
<button data-target="#tarid"></button>

<!-- Collapsible element -->
<div class="collapse" id="tarid">
    <p>Content in this element will be collapsed by default and expand when its controller is toggled</p>
</div>

<!-- Link tags may also be used as controllers, href must be set as a no-script fallback and data-target must be set for the element to be selected -->
<a href="#second-collapse" data-target="#second-collapse">Expand collapsed content</a>

<!-- Using the .in class will cause a section to be expanded by default -->
<div class="collapse in" id="second-collapse">
    <p>This element is expanded by default and will collapse when its controller is toggled</p>
</div>
```

So, in summary:
**Controllers:** must be a button or link with the attribute `data-target` that specifies which container it controls. If the controller is a link, href _must_ match `data-target` in order to validate and as a fallback if JavaScript is disabled

**Collapsible Containers:** must have the `.collapse` class and a unique ID that will match its controller's `data-target` attribute

**Comment by brharp** [comment]
I had to add ``data-toggle=collapse`` to Nevan's latest example to get it to work.

**Comment by nevantan** [comment]
For some reason, the `href` attribute on the link in my example is having the # stripped out (leaving it with `<a href="second-collapse">` rather than `<a href="#second-collapse">`) which causes it to link to another page rather than triggering the collapse.
